### DISCUSSION TITLE
About thread safety

### DISCUSSION THREAD

**Discussion by Xyncgas** [nlohmann/json#2366] [Q&A] [answered]
`json a;`
Thread 1:a["a1]="yes";
Thread 2:a["a2]="no";
Thread 3:a["a3]="yes";
Thread 4:a["a4]="no";
Thread 5:a["a5]="yes";

is there a mutex built-in to guarantee the underlaying data structure not broken and modified with mutex?

**Comment by nlohmann** [comment, marked as answer]
No, the container is like a map or a vector: you have to ensure thread safety yourself.

**Comment by Xyncgas** [comment]
ahh ty. I asked because I felt like as if there's a mutex, when I compiled it to webassembly and used multithreading and I had to be sure yea thanks
